test/mox_test.exs

```diff
@@ -351,39 +351,84 @@ defmodule MoxTest do
 
       verify!()
       expect(CalcMock, :add, fn x, y -> x + y end)
+      expect(SciCalcOnlyMock, :exponent, fn x, y -> x * y end)
 
-      message = ~r"expected CalcMock.add/2 to be invoked once but it was invoked 0 times"
-      assert_raise Mox.VerificationError, message, &verify!/0
+      assert_raise(Mox.VerificationError, &verify!/0)
+      try do
+        verify!()
+      rescue
+        error in Mox.VerificationError ->
+          assert error.message =~ ~r"expected CalcMock.add/2 to be invoked once but it was invoked 0 times"
+          assert error.message =~ ~r"expected SciCalcOnlyMock.exponent/2 to be invoked once but it was invoked 0 times"
+      end
```

Comment:
```suggestion
      assert %Mox.VerificationError{} = exception = catch_error(&verify!/0)
      assert error.message =~ ~r"expected CalcMock.add/2 to be invoked once but it was invoked 0 times"
      assert error.message =~ ~r"expected SciCalcOnlyMock.exponent/2 to be invoked once but it was invoked 0 times"
```

Reply:
If it works, can you please apply this style in other places to clean up the tests? Thanks!

Reply:
Ah! Great call. The `catch_error/1` method didn't work because [it only catches assertion errors](https://github.com/elixir-lang/elixir/blob/v1.12.3/lib/ex_unit/lib/ex_unit/assertions.ex#L881-L891) but seeing that syntax got me re-reading the documentation & I realized that `assert_raise/2` explicitly "[Returns the rescued exception](https://hexdocs.pm/ex_unit/1.12.3/ExUnit.Assertions.html#assert_raise/2)" which also enables the style of refactor you've proposed! 

Implemented the improvements in 1ee925f7051fa475b2063ed2ebd1f6304921326b and 999810cfba96b31b6edefd2b9e7f346072c5cfbb